pyvrp/cpp/Solution.cpp

```diff
@@ -116,9 +193,28 @@ Solution::Solution(ProblemData const &data,
     }
 
     std::vector<size_t> visits(data.numClients() + 1, 0);
+    std::vector<size_t> used_vehicles(data.numVehicleTypes(), 0);
     for (auto const &route : routes)
+    {
+        if (route.empty())
+        {
+            std::ostringstream msg;
+            msg << "Solution should not contain empty routes.";
+            throw std::runtime_error(msg.str());
+        }
```

Comment:
I think supporting empty routes adds very little if nothing but leaves us with more edge cases to check:
* Are two solutions considered equal if one adds some empty routes? If not, then we should check 'used' (but not really) vehicle type counts in `Solution::operator==`.
* Do we guarantee that loading and exporting a solution into local search, without doing search gives the same 'solution' (if that is defined to be different with empty routes). Same for using a Solution constructor and `get_routes()` roundtrip.
* How should SREX handle empty routes?

Since I see no benefit, I think it's better to just guarantee we have no empty routes, which makes our implementations simpler as we can forget about these edge cases.

Reply:
OK, but then we can simplify a lot more.